airflow/providers/cncf/kubernetes/executors/kubernetes_executor.py

```diff
@@ -177,6 +177,37 @@ def _make_safe_label_value(self, input_value: str | datetime) -> str:
             return pod_generator.datetime_to_label_safe_datestring(input_value)
         return pod_generator.make_safe_label_value(input_value)
 
+    def get_pod_labels_combined_str_to_pod_map(self) -> dict[str, k8s.V1Pod]:
+        """
+        List the worker pods owned by this scheduler and create a map containing pod combined labels search str -> pod.
+
+        For every pod, it creates two below entries in the map
+        dag_id={dag_id},task_id={task_id},airflow-worker={airflow_worker},<map_index={map_index}>,run_id={run_id}
+        """
+        # airflow worker label selector batch call
+        kwargs = {"label_selector": f"airflow-worker={self._make_safe_label_value(str(self.job_id))}"}
+        if self.kube_config.kube_client_request_args:
+            kwargs.update(self.kube_config.kube_client_request_args)
+        pod_list = self._list_pods(kwargs)
+
+        # create a set against pod query label fields
+        pod_labels_combined_str_to_pod_map = {}
+        for pod in pod_list:
+            dag_id = pod.metadata.labels.get("dag_id", None)
+            task_id = pod.metadata.labels.get("task_id", None)
+            airflow_worker = pod.metadata.labels.get("airflow-worker", None)
+            map_index = pod.metadata.labels.get("map_index", None)
+            run_id = pod.metadata.labels.get("run_id", None)
+            if dag_id is None or task_id is None or airflow_worker is None:
+                continue
+            label_search_base_str = f"dag_id={dag_id},task_id={task_id},airflow-worker={airflow_worker}"
+            if map_index is not None:
+                label_search_base_str += f",map_index={map_index}"
+            if run_id is not None:
```

Comment:
Where'd the execution_date case go?

Reply:
execution_date is deprecated and removed in this PR. Do we still need to support it?

Reply:
Probably not, but we should probably add a note to the release notes that upgrading from whatever version set those no longer works if you have running tasks.

That also likely triggers a major version. Not sure it's worth it just to remove that deprecated code. cc @eladkal

Reply:
Did we raise warning for this before?

Reply:
No, execution_date has been deprecated and removed from XComs and other components in Airflow 3. In Airflow 2, execution_date is no longer used in pod labels; it was only kept to manage pods during the migration from Airflow 1 to 2.

Reply:
In that case I think it's fine to make major release with just a warning note in the change log explaning what was removed and the impact (which according to you is minimal if any?)
Please set versions in provider.yaml to `9.0.0` and in CHANGELOG.rst add

```
main
.....

.. warning::
  Explain your change and impact
```

Reply:
Cool, if we are doing a major, now is the time to remove any other deprecations we have stacked up. @dirrao have at it :)

Reply:
Done.